codegen/src/main/java/software/amazon/awssdk/codegen/poet/rules/EndpointResolverInterceptorSpec.java

```diff
@@ -106,6 +115,13 @@ private MethodSpec modifyRequestMethod() {
         b.beginControlFlow("try");
         b.addStatement("$T result = $N.resolveEndpoint(ruleParams(context, executionAttributes)).join()", Endpoint.class,
                        providerVar);
```

Comment:
Not sure how that would happen, since the HostPrefix logic requires access to the request params which may not be part of the set of rules parameters.

Reply:
Sounds like the endpoint parameters need to be informed by more than just the rules, then... I know I'm just complaining about the endpoints 2.0 design and not this PR, so feel free to ignore this comment, then.

Reply:
> Sounds like the endpoint parameters need to be informed by more than just the rules, then

Yep, exactly!